### PR TITLE
Improve startup

### LINKED ISSUE #1845: no Trackers detected in a single set home page @FFox
Hey there, 

maybe this is a problem or intended functionality?

If I set a single webpage as my Firefox "home page" to load it directly after start up PB is showing a green Status with 0. Looking into PB it's telling me "Hurray, no Trackers detected" and there is no list of domains in PB. Looking at my startup page  I see every single ad - nothing is handled or blocked even if I manually specified some domains to be blocked.
Do I start FFox with an empty page and then visit only the page I set as startup page PB is turning red seeing a lot of domains to block. Opening PB I can see all settings of PB itself and manualy changed by me. A lot of the ads are blocked. 

I do not see why there snhould be a difference between the home page and directly opening that page . It feels like "home page" is completely whitelisted.

Recognized in Firefox 58, I think it happend to FFox 57.x , too.
PB Version 2018.1.22

Best regards
Alex.

### PR DESCRIPTION
**Pull request by ghostwords**
Related to #1845.

- [x] Fixes `badger.INITIALIZED` to account for asynchronous initialization tasks (such as seed data loading). This removes the need for workarounds in places that care about Badger being fully initialized (might want to keep [them though in badger-sett](https://github.com/EFForg/badger-sett/commit/06d40f03304c4670004c18aecb3688ae192c1fcf#diff-ee054b4b797f6c2a5743d2654764a8edR263-R266) for a while longer to enable running with Badgers that won't have this fix).
- [x] Postpones non-essential initialization tasks (such as fetching yellowlist/DNT hashes from eff.org)
- [ ] ~~Sets up an early `chrome.onBeforeRequest` listener to cancel all requests before Badger is done initializing. Reloads the tabs for the cancelled requests upon Badger being ready. (fb6876642e4995c80c9a579296a53e8571cdd0fc, 0de95d89f37cd74d66dadf5baaf26c8e19193203) This should be guaranteed to catch all requests on startup in Firefox thanks to a ["persistent" listener](https://github.com/mdn/sprints/issues/1015), and should catch more requests on startup in Chrome thanks to initializing a blocking listener faster.~~

### CODE REVIEW

File: src/js/webrequest.js [hunk 1]
```diff
@@ -855,15 +856,55 @@ function dispatcher(request, sender, sendResponse) {
...
+
+/**
+ * Cancels all requests, populates list of pending tabs to be reloaded later,
+ * when Privacy Badger is ready.
+ *
+ * @param {Object} details chrome.webRequest request details
+ * @returns {Object}
+ */
+function onBeforeReady(details) {
+  if (!pendingTabs || details.tabId < 0) {
```
Comment (bcyphers) [inline src/js/webrequest.js]: when would pendingTabs be falsey here?

Comment (ghostwords) [reply src/js/webrequest.js]: We set `pendingTabs` to `null` at the end of `reloadPendingTabs()`. Is it possible to still have requests queued for `onBeforeReady()` after `stopBeforeReadyListener()` is called? I would have to look into it.

Comment (ghostwords) [reply src/js/webrequest.js]: We may want to ignore "main_frame" resources to avoid running into https://github.com/gorhill/uBlock/issues/3130

Comment (ghostwords) [reply src/js/webrequest.js]: Should the listener not be removed for it to be treated as "persistent" by Firefox? See https://github.com/uBlockOrigin/uBlock-issues/issues/128#issuecomment-435594531

File: src/js/webrequest.js [hunk 2]
```diff
@@ -887,17 +905,57 @@ function dispatcher(request, sender, sendResponse) {
...
+}
+
+
 /*************** Event Listeners *********************/
+function startBeforeReadyListener() {
+  chrome.webRequest.onBeforeRequest.addListener(onBeforeReady, {urls: ["<all_urls>"]}, ["blocking"]);
+}
+
+function stopBeforeReadyListener() {
+  chrome.webRequest.onBeforeRequest.removeListener(onBeforeReady);
```
Comment (ghostwords) [inline src/js/webrequest.js]: Should the listener not be removed for it to be treated as "persistent" by Firefox? See https://github.com/uBlockOrigin/uBlock-issues/issues/128#issuecomment-435594531

**Pull request merged**